src/binder/bind/bind_graph_pattern.cpp

```diff
@@ -179,8 +179,11 @@ void Binder::bindQueryRel(const RelPattern& relPattern,
                                   " are not connected through rel " + parsedName + ".");
         }
     }
-    auto dataType = isVariableLength ? common::LogicalType(common::LogicalTypeID::RECURSIVE_REL) :
-                                       common::LogicalType(common::LogicalTypeID::REL);
+    auto dataType = isVariableLength ?
```

Comment:
Can u check other places where we switch for RECURSIVE_REL logical type. We should propogate child type to those places as well.

For example
```
AuxiliaryBufferFactory::getAuxiliaryBuffer
```